libcalico-go/lib/backend/model/policy.go

```diff
@@ -127,3 +127,59 @@ func (p Policy) String() string {
 	}
 	return strings.Join(parts, ",")
 }
+
+// buildLegacyPolicyKey builds a legacy policy key from the given name.
+// Legacy policy key names included the namespace and name of the policy, with
+// hints of the kind that can be used to reconstruct the full key.
+//
+// Legacy policy keys will be sent by Typha versions prior to v3.32.0.
+func buildLegacyPolicyKey(name string) Key {
+	// First, split based on "/" to see if we have a namespace.
+	parts := strings.SplitN(name, "/", 2)
+	if len(parts) == 2 {
+		// We have a namespace. This can be one of a few different kinds:
+		// - knp.default.<name>          -> KubernetesNetworkPolicy
+		// - staged:<name>               -> StagedNetworkPolicy
+		// - staged:knp.default.<name>   -> StagedKubernetesNetworkPolicy
+		// - <name>                      -> NetworkPolicy
+		namespace := parts[0]
```

Comment:
Do we need to worry about openstack security groups here?  Looks like they have prefix `ossg.default.`

Reply:
I think a different kind makes sense. Worth discussing with @nelljerram I think.

Reply:
My 2p: I think having a separate kind would be consistent; it's what we're dong for the other derived kinds.